android-agent/src/main/kotlin/io/opentelemetry/android/agent/OpenTelemetryRumInitializer.kt

```diff
@@ -79,9 +80,17 @@ object OpenTelemetryRumInitializer {
         instrumentations?.let { configure ->
             InstrumentationConfiguration(rumConfig).configure()
         }
+
+        // ensure we're using the Application Context to prevent potential leaks
+        val ctx =
+            when (context) {
+                is Application -> context
+                else -> context.applicationContext
```

Comment:
```suggestion
                else -> context.applicationContext ?: context
```

applicationContext can be nullable in very specific cases

Reply:
IMO it is reasonable to document that as a caveat of this approach, specifically that if you attempt to initialize within `attachBaseContext` that lifecycle callbacks won't work. At Embrace we cast applicationContext to `Application` when setting lifecycle callbacks & anecdotally haven't experienced customers running into this issue. I do know that some folks want to initialize their SDK earlier than the `Application` lifecycle and I view it as a tradeoff that allows them to capture early telemetry that might be more important to them than other telemetry that might become harder to capture because of the change.

Reply:
I've altered the code with your suggestion and tweaked the docs to indicate what could happen if a library consumer passed in the wrong context